kernel/src/fs/configfs/fs.rs

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::sync::Arc;
+
+use aster_block::BlockDevice;
+use aster_systree::SysBranchNode;
+
+use super::inode::ConfigInode;
+use crate::{
+    fs::{
+        configfs::systree_node::ConfigRootNode,
+        registry::{FsProperties, FsType},
+        utils::{systree_inode::SysTreeInodeTy, FileSystem, FsFlags, Inode, SuperBlock},
+        Result,
+    },
+    prelude::*,
+};
+
+/// A file system that can act as a manager of kernel objects
+pub struct ConfigFs {
+    sb: SuperBlock,
+    root: Arc<dyn Inode>,
+    systree_root: Arc<ConfigRootNode>,
+}
+
+// Magic number for configfs (taken from Linux)
+const MAGIC_NUMBER: u64 = 0x62656570;
+const BLOCK_SIZE: usize = 4096;
+const NAME_MAX: usize = 255;
+
+impl ConfigFs {
+    pub(super) fn new(root_node: Arc<ConfigRootNode>) -> Arc<Self> {
+        let sb = SuperBlock::new(MAGIC_NUMBER, BLOCK_SIZE, NAME_MAX);
+        let root_inode = ConfigInode::new_root(root_node.clone());
+
+        Arc::new(Self {
+            sb,
+            root: root_inode,
+            systree_root: root_node,
+        })
+    }
+
+    pub(super) fn systree_root(&self) -> &Arc<ConfigRootNode> {
+        &self.systree_root
+    }
+}
+
+impl FileSystem for ConfigFs {
+    fn sync(&self) -> Result<()> {
+        // ConfigFs is volatile, sync is a no-op
+        Ok(())
+    }
+
+    fn root_inode(&self) -> Arc<dyn Inode> {
+        self.root.clone()
+    }
+
+    fn sb(&self) -> SuperBlock {
+        self.sb.clone()
+    }
+
+    fn flags(&self) -> FsFlags {
+        FsFlags::empty()
+    }
+}
+
+pub(super) struct ConfigFsType {
+    systree_root: Arc<ConfigRootNode>,
+}
+
+impl ConfigFsType {
+    pub(super) fn new(systree_root: Arc<ConfigRootNode>) -> Arc<Self> {
+        Arc::new(Self { systree_root })
+    }
+}
+
+impl FsType for ConfigFsType {
+    fn name(&self) -> &'static str {
+        "configfs"
+    }
+
+    fn properties(&self) -> FsProperties {
+        FsProperties::empty()
+    }
+
+    fn create(
+        &self,
+        _args: Option<CString>,
+        _disk: Option<Arc<dyn BlockDevice>>,
+        _ctx: &Context,
+    ) -> Result<Arc<dyn FileSystem>> {
+        if super::CONFIG_SINGLETON.is_completed() {
+            return_errno_with_message!(Errno::EBUSY, "the configfs has been created");
```

Comment:
I think both this check and the one in `SysFsType::create` are wrong.

https://github.com/asterinas/asterinas/blob/3b606f5b6c127c1a5631d8c2c652d23b9a8f9937/kernel/src/syscall/mount.rs#L155-L184

The user space is allowed to mount a ProcFS, SysFS, or ConfigFS multiple times; for example, when each container will mount these special FSes. This means that the corresponding `FsType::create` may be invoked multiple times. So this function should simply return the FS singleton if it is already initialized.